
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>inmembrane</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">

    <!-- Le styles -->
    <link href="css/bootstrap.css" rel="stylesheet">
    <style type="text/css">
      body {
        padding-top: 20px;
        padding-bottom: 40px;
      }

      /* Custom container */
      .container-narrow {
        margin: 0 auto;
        max-width: 700px;
      }
      .container-narrow > hr {
        margin: 30px 0;
      }

      /* Main marketing message and sign up button */
      .jumbotron {
        margin: 60px 0;
        text-align: center;
      }
      .jumbotron h1 {
        font-size: 72px;
        line-height: 1;
      }
      .jumbotron .btn {
        font-size: 21px;
        padding: 14px 24px;
      }

      /* Supporting marketing content */
      .marketing {
        margin: 60px 0;
      }
      .marketing p + h4 {
        margin-top: 28px;
      }
    </style>
    <link href="css/bootstrap-responsive.css" rel="stylesheet">

    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <link rel="shortcut icon" href="ico/favicon.ico">
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="ico/apple-touch-icon-144-precomposed.png">
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="ico/apple-touch-icon-114-precomposed.png">
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="ico/apple-touch-icon-72-precomposed.png">
    <link rel="apple-touch-icon-precomposed" href="ico/apple-touch-icon-57-precomposed.png">
  </head>

  <body>

    <a href="https://github.com/boscoh/inmembrane"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"></a>

    <div class="container-narrow">

      <div class="masthead">
        <ul class="nav nav-pills pull-right">
          <li><a href="index.html">Home</a></li>
          <li class="active"><a href="#">Configuration</a></li>
          <li><a href="api.html">API</a></li>
        </ul>
        <h3 class="muted">inmembrane</h3>
      </div>

      <hr>

      <div class="row-fluid marketing">
        <div class="span12">
         
         
<h1>Configuration of <em>inmembrane</em></h1>
<p>By default, <em>inmembrane</em> will run the Gram+ protocol using remote web services for the analysis (and HMMER locally) without requiring any additional configuration.
</p>
<p>On the first use of the program, a default <code>inmembrane.config</code> configuration file will be generated. Once created, you can edit this <code>inmembrane.config</code>. Subsequent use of the program will reference the existing <code>inmembrane.config</code>. This file contains all the user adjustable parameters (as a Python dictionary). To run the Gram- protocol, use locally installed copies of SignalP/LipoP/TMHMM or tweak the external loop threshold, you may want to edit the configuration file to your needs.
</p>
<p>The default parameters are as following, and they will be explained in sections below:
</p>
<pre><code>{
  'fasta': '',
  'csv': '',
  'out_dir': '',
  'protocol': 'gram_pos', # 'gram_neg'

#### Signal peptide and transmembrane helix prediction
  'signalp4_bin': '',
# 'signalp4_bin': 'signalp',
  'lipop1_bin': '',
# 'lipop1_bin': 'LipoP',
  'tmhmm_bin': '',
#  'tmhmm_bin': 'tmhmm',
  'memsat3_bin': 'runmemsat',
  'helix_programs': ['tmhmm'],
# 'helix_programs': ['tmhmm', 'memsat3'],
  'terminal_exposed_loop_min': 50, # unused in gram_neg protocol
  'internal_exposed_loop_min': 100, # try 30 for gram_neg

#### Sequence similarity and motif prediction
  'hmmsearch3_bin': 'hmmsearch',
  'hmm_profiles_dir': '%(hmm_profiles)s',
  'hmm_evalue_max': 0.1,
  'hmm_score_min': 10,

#### Outer membrane beta-barrel predictors
  'barrel_programs': ['bomp', 'tmbetadisc-rbf'],
# 'barrel_programs': ['bomp', 'tmbetadisc-rbf', 'tmbeta'],
  'bomp_clearly_cutoff': 3, # &gt;= to this, always classify as an OM(barrel)
  'bomp_maybe_cutoff': 1, # must also have a signal peptide to be OM(barrel)
  'tmbetadisc_rbf_method': 'aadp', # aa, dp, aadp or pssm
}
</code></pre><p>Existing <code>inmembrane.config</code> files can be changed. Since this file is in fact just Python code, normal Python syntax rules apply. Everything after a hash (#) character on a line is ignored - these are used as comments, and to used indicate alternative configurations that you might want to use.
</p>
<p>The parameters will be explained below, grouped in terms of related parameters.
</p>

<h3>General Parameters</h3>
<p>The input to <em>inmembrane</em> is a FASTA formatted set of protein sequences. There are currently two different protocols (or 'workflows') - one for Gram+ bacteria (the default), and one for Gram- bacteria. The results are saved to a CSV file as an output.
</p>
<ul>
 <li>
     <code>fasta</code>: the pathname of the FASTA file holding the protein sequences
 </li>

 <li>
     <code>protocol</code>: switches between different protocols for analysing the proteomes, currently you can choose between <code>gram_pos</code> and <code>gram_neg</code>.
 </li>

 <li>
     <code>csv</code>: the name of the output file in CSV (Excel-compatible) format. Defaults to a file using the same base filename as 'fasta'
 </li>
</ul>
<p>Both <code>fasta</code> and <code>csv</code> should usually be left blank by default, since the input file is usually specified directly on the commandline.
</p>
<p>For potential debugging, it's important that all intermediate files are saved, and they are saved in: 
</p>
<ul>
 <li>
     <code>out_dir</code>: the directory where all intermediate output files are located. Defaults to the same filename as <code>fasta</code> (without the .fasta extension)
 </li>
</ul>

<h3>Optional - Locally installed binaries or remote web services</h3>
<p>One of the problems that <em>inmembrane</em> tackles is the complexity of using and installing multiple pieces of sequence analysis software. For older programs, the only option was to install the local binaires of all the external programs. This raises problems involved with getting the binary (or source) for the right platform, compilation with the right tools, and installation and configuration in the correct location. In contrast, many of these tools now provide a web-interface. In <em>inmembrane</em>, we have added facilities to use remote web services, when provided, whether through a RESTful interface, or through a SOAP interface.
</p>
<p>By default, <em>inmembrane</em> will use web services, specified by setting the <code>*_bin</code> parameter as empty string. To use a local binary binary version of a program, the appropriate <code>*_bin</code> parameter must be set to the name (or full path) of the executable.
</p>
<ul>
 <li>
     <code>signalp4_bin</code>, <code>lipop1_bin</code>, <code>tmhmm_bin</code>, <code>memsat3_bin</code>: the full pathname of the binaries used for the analysis. If empty, the web version is used instead.
 </li>
</ul>

<h3>The Potentially Surface Exposed (PSE) Algorithm</h3>
<p>The key part of the surface-exposure prediction evaluates the length of loops in the extramembrane parts of membrane proteins. This analysis makes most sense for Gram+ bacteria with a single membrane, where any membrane protein can potentially expose loops to the extracellular environment. This evaluation requires the key parameter of loop cutoff that determines the minimum length of a loop that is required to potentially pass through the glycan layer. 
</p>
<p>In the original <em>SurfG+</em> paper <a href="http://dx.doi.org/10.1002/pmic.200800195">(Barinov et al. 2009. Proteomics 9:61-73)</a>, it was found that a linear polypeptide length of ~50 amino acids most closely matched the length able to traverse the cell wall and become succeptible to protease shaving in the model Gram+ bacterium studied. Therefore for internal loops between transmembrane segments, this gives an accessible length threshold of ~100 amino acids, which allows the loop to pass through the cell-wall layer and back down to the membrane again.
</p>
<ul>
 <li>
     <code>terminal_exposed_loop_min</code>: this is the length (in amino acids) given to define an N- or C- terminal segment of a protein that is long enough to stick out of the bacterial cell-wall to be considered surface-exposed
 </li>

 <li>
     <code>internal_exposed_loop_min</code>: the length of a loop between two membrane embedded segments able to traverse the cell wall and present as surface exposed
 </li>
</ul>
<p>These values should be modified based on the bacterial species (based on cell wall thickness) and type of experiment being used to determine surface exposure (protease shaving vs. epitope mapping vs. chemical crosslinking).
</p>
<p>There are several different ways to identify proteins with a subcellular localization that may display loops that protruding outside the glycan cell-wall of the bacteria. These will be discussed in the following sections.
</p>

<h3>Proteins attached to the Glycan Cell-wall and the Lipid bilayer</h3>
<p>The clearest example of proteins that may protrude outside of the Gram+ glycan layer are proteins that possess known cell-wall attachment motifs. Motifs for domains known to bind to the glycan cell-wall are taken from PFAM as HMMER profiles, and are included in the <em>inmembrane</em> installation. For the Gram+ protocol, these are: CW_binding_1, Gram_pos_anchor, GW2, LPxTG, LysM, PG_binding_1, PG_binding_3, CW_binding_2, GW1, GW3, LPxTG_PS50847, NLPC_P60, PG_binding_2 and SLH.
</p>
<p>HMMER is used to search the profiles against the input query sequences. The cutoffs for the identification of a motif are given by <code>hmm_evalue_max</code> and <code>hmm_score_min</code>.
</p>
<ul>
 <li>
     <code>hmmsearch3_bin</code>: the binary for HMMER
 </li>

 <li>
     <code>hmm_evalue_max</code>: the maximum expectation value cutoff for acceptable prediction in HMMER
 </li>

 <li>
     <code>hmm_score_min</code>: the minimum score cutoff for acceptatble prediction in HMMER
 </li>
</ul>
<p>Another class of proteins associated with the membrane surface membrane are lipoproteins, which are posttranlationally modified to include a covalently attached N-terminal lipid moeity. LipoP is used to identify the N-terminal lipidation motif, found after a secretion signal. 
</p>
<ul>
 <li>
     <code>lipop_bin</code>: the location of the binary of LipoP. If this is given as '' (default), then <em>inmembrane</em> uses the web service at <a href="http://www.cbs.dtu.dk/services/LipoP/">http://www.cbs.dtu.dk/services/LipoP/</a>.
 </li>
</ul>

<h3>Transmembrane-&alpha;-helix Predictors</h3>
<p>The other main class of protein in Gram+ proteins with extracellular loops are integral membrane proteins containing transmembrane &alpha;-helices. The loops between transmembrane segments, as well as N-terminal and C-terminal non-membrane embedded regions are all candidates to be exposed outside the glycan layer.
</p>
<p>There is no <em>de facto</em> standard transmembrane helix predictor, and thus <em>inmembrane</em> has been written to allow pluggable options for transmembrane helix prediction. To allow for this, <em>inmembrane</em> allows the results of more than one helix predictor, and takes a greedy approach to annotating potentially surface exposed proteins - the prediction that gives the longest extracellular loops will be used. The parameter 'helix_programs' lists all the transmembrane helix predictors to be used.
</p>
<ul>
 <li>
     <code>helix_programs</code>: a list of transmembrane-helix predictors to be used. Currently, <em>inmembrane</em> can handle <code>tmhmm</code> and <code>memsat3</code>.
 </li>
</ul>

<h3>&beta;-barrel predictors</h3>
<p>These parameters only apply to the Gram- protocol, since Gram+ bacteria contain only a single membrane containing helical integral membrane proteins and not outer membrane beta barrels (OMPbb's) (although it's worth noting that things may be so simple, since some species of Mycobacterium may contain outer membranes and associated OMPs <a href="http://dx.doi.org/10.1016/j.tube.2008.02.004">http://dx.doi.org/10.1016/j.tube.2008.02.004</a> ). By default, <em>inmembrane</em> uses BOMP (http://services.cbu.uib.no/tools/bomp/) and TMBETADISC-RBF (http://rbf.bioinfo.tw/~sachen/OMPpredict/TMBETADISC-RBF.php).
</p>
<ul>
 <li>
     <code>barrel_programs</code>: a list of web services for outer membrane &amp;beta-barrel prediction ['bomp', 'tmbetadisc-rbf'],
 </li>

 <li>
     <code>bomp_clearly_cutoff</code>: 3, # if greater than or equal to this, always classify as an OM(barrel)
 </li>

 <li>
     <code>bomp_maybe_cutoff</code>: 1, # must also have a signal peptide to be OM(barrel)
 </li>

 <li>
     <code>tmbetadisc_rbf_method</code>: 'aadp', # TMBETADISC-RBF method: aa (single amino acid composition based), dp (dipeptide composition based), aadp (both aa and dp) or pssm (position specific mutation matrix based)
 </li>
</ul>
<p>A plugins for TMBETA (<code>tmbeta</code>) is available but disabled by default since the general prediction accuracy or web service reliability was found to be lacking during testing. Previously, <em>inmembrane</em> also provided a plugin for TMBHUNT (<code>tmbhunt</code>), however this has been removed since the web service is no longer active.
</p>
<p>Since topology prediction for arbitrary OMPbb sequences is still very unreliable, the Gram- protocol does not attempt to do loop length prediction but simply annotates potential OMPbb proteins as such.
</p>


        </div>
      </div>

      <hr>

      <div class="footer">
        <p>&copy; 2012, Andrew Perry and Bosco Ho</p>
      </div>

    </div> <!-- /container -->

    <!-- Le javascript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="js/jquery.js"></script>
    <script src="js/bootstrap-transition.js"></script>
    <script src="js/bootstrap-alert.js"></script>
    <script src="js/bootstrap-modal.js"></script>
    <script src="js/bootstrap-dropdown.js"></script>
    <script src="js/bootstrap-scrollspy.js"></script>
    <script src="js/bootstrap-tab.js"></script>
    <script src="js/bootstrap-tooltip.js"></script>
    <script src="js/bootstrap-popover.js"></script>
    <script src="js/bootstrap-button.js"></script>
    <script src="js/bootstrap-collapse.js"></script>
    <script src="js/bootstrap-carousel.js"></script>
    <script src="js/bootstrap-typeahead.js"></script>

  </body>
</html>
