
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>inmembrane</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">

    <!-- Le styles -->
    <link href="css/bootstrap.css" rel="stylesheet">
    <style type="text/css">
      body {
        padding-top: 20px;
        padding-bottom: 40px;
      }

      /* Custom container */
      .container-narrow {
        margin: 0 auto;
        max-width: 700px;
      }
      .container-narrow > hr {
        margin: 30px 0;
      }

      /* Main marketing message and sign up button */
      .jumbotron {
        margin: 60px 0;
        text-align: center;
      }
      .jumbotron h1 {
        font-size: 72px;
        line-height: 1;
      }
      .jumbotron .btn {
        font-size: 21px;
        padding: 14px 24px;
      }

      /* Supporting marketing content */
      .marketing {
        margin: 60px 0;
      }
      .marketing p + h4 {
        margin-top: 28px;
      }
    </style>
    <link href="css/bootstrap-responsive.css" rel="stylesheet">

    <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <link rel="shortcut icon" href="ico/favicon.ico">
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="ico/apple-touch-icon-144-precomposed.png">
    <link rel="apple-touch-icon-precomposed" sizes="114x114" href="ico/apple-touch-icon-114-precomposed.png">
    <link rel="apple-touch-icon-precomposed" sizes="72x72" href="ico/apple-touch-icon-72-precomposed.png">
    <link rel="apple-touch-icon-precomposed" href="ico/apple-touch-icon-57-precomposed.png">
  </head>

  <body>

    <a href="https://github.com/boscoh/inmembrane"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"></a>

    <div class="container-narrow">

      <div class="masthead">
        <ul class="nav nav-pills pull-right">
          <li class="active"><a href="#">Home</a></li>
          <li><a href="customization.html">Configuration</a></li>
          <li><a href="api.html">API</a></li>
        </ul>
        <h3 class="muted">inmembrane</h3>
      </div>

      <hr>

      <div class="jumbotron">
        <h1>inmembrane</h1>
        <p class="lead">Prediction of surface-exposed proteins in bacterial cells</p>
        <a class="btn btn-large btn-success" href="http://pypi.python.org/packages/source/i/inmembrane/inmembrane-0.94.1.tar.gz">Download</a> <br/>
        (stable release, v0.94)
      </div>

      <hr>

      <div class="row-fluid marketing">
        <div class="span12">

          
<h2>What is <em>inmembrane</em>?</h2>
<p><em>inmembrane</em> is a program to predict if a set of proteins are likely to be surface-exposed in a bacterial cell based on their predicted subcellular localization and membrane topology. It is designed to be relatively easy to install, with options for running the analysis using web services and/or locally installed sequence analysis packages.
</p>
<p><em>inmembrane</em> runs on Linux, OS X and other Unix-based systems. It has been tested on Ubuntu Linux (11.04+) and Mac OS X (10.6+). It hasn't been tested on Windows, and is unlikely to run without changes.
</p>

<h3>Quickstart for the bold &amp; impatient</h3>
<pre><code>$ sudo pip install inmembrane
$ inmembrane_scan --test
$ inmembrane_scan my_gram_pos_proteome.fasta
</code></pre><p><em>(Some tests may fail if you don't have various external binaries installed, but don't despair - the only external program <strong>required</strong> by is HMMER v3, everything else uses a web service in the default configuration)</em>
</p>

<h3>How can inmembrane help you?</h3>
<p>We envisage <em>inmembrane</em> being of use in three ways:
</p>
<ol>
 <li><p>Quirk and dirty annotation of a proteome in order to cross-check your experiment (eg, gaining an overview of hits from a gram-positive bacterial cell-surface shaving experiment).
</p>

 </li>

 <li><p>You want to tweak your own rule-based method for subcellular localization assignment from sequence. Check out our <a href="customization.html">parameterization guide</a>.
</p>

 </li>

 <li><p>You want to make your own protein sequence analysis work-flow, and would like to use <em>inmembrane</em> as a useful Python starting point. Have a look at our <a href="api.html">programming model and API guide</a>.
</p>

 </li>
</ol>

<h3>Combining analyses into a workflow</h3>
<p>The cell-surface exposure algorithm used by the <em>inmembrane</em> 'gram-positive' protocol is based on the method described in <a href="http://dx.doi.org/10.1002/pmic.200800195">(Barinov et al. 2009. Proteomics 9:61-73)</a>. The workflow involves annotating the provided protein sequences using a number of external programs and/or web services providing:
</p>
<ul>
 <li>
     transmembrane helix prediction (eg TMHMM)
 </li>

 <li>
     lipoprotein prediction (eg LipoP)
 </li>

 <li>
     secretion signal prediction (eg SignalP)
 </li>

 <li>
     cell-wall binding motif prediction (eg via HMMER)
 </li>
</ul>
<p><em>inmembrane</em> simplifies collating the output and presenting an overview which allows biologists to immediately gain insight into the likely localization and surface exposure of their bacterial proteome of interest.
</p>

<h2>Using <em>inmembrane</em></h2>

<h3>Installation</h3>
<p>Installing and running <em>inmembrane</em> requires:
</p>
<ul>
 <li>
     Python: it should be on most modern systems, but if not, go to <a href="http://www.python.org">http://www.python.org</a>
 </li>

 <li>
     pip: <a href="http://pypi.python.org/pypi/pip">http://pypi.python.org/pypi/pip</a>
 </li>

 <li>
     HMMER: this is the only sequence analysis software that is required to be locally installed <a href="http://hmmer.janelia.org/software">http://hmmer.janelia.org/software</a>
 </li>
</ul>
<p>On Debian/Ubuntu systems, you can install the minimal dependencies using:
</p>
<pre><code>$ sudo apt-get install python-pip hmmer
</code></pre><p>Once these are installed on your system, <em>inmembrane</em> can be installed with:
</p>
<pre><code>$ sudo pip install inmembrane
</code></pre>
<h3>Running <em>inmembrane</em></h3>
<p>Let's say you have all the protein sequences you want to analyze in a FASTA text file called <code>my_proteome.fasta</code>.
</p>
<p><em>Inmembrane</em> is run from the command line like:
</p>
<pre><code>$ inmembrane_scan my_proteome.fasta
</code></pre><p>Assuming the web services <em>inmembrane</em> relies on are available, the analysis should complete within seconds or minutes.
</p>
<p>A summary is printed to the terminal, and the results will also be written to <code>my_proteome.csv</code>, which can be opened by a spreadsheet application such as Microsoft Excel. Citations for all external programs and services are written to <code>my_proteome/citations.txt</code> - this is very handy and important when it comes to writing up a publication using the results of <em>inmembrane</em> and the software it orchestrates.
</p>
<p><strong>Pro-tip:</strong> The directory created by <em>inmembrane</em> (eg named <code>my_proteome</code>), as well as containing the <code>citations.txt</code> list, also contains cached output files from the individual external analyses, the original configuration file and the input FASTA file. If you run <em>inmembrane</em> a second time on the same FASTA file, the results will be read from this directory rather than re-running each external analysis program. To re-run the analysis from scratch (with say, different parameters), you must remove (or rename) this directory. To re-run just part of the analysis you can remove individual files from this directory.
</p>

<h3>Interpreting the output</h3>
<p>It should look something like this: 
</p>
<pre><code>SPy_0008  CYTOPLASM(non-PSE)  .                         SPy_0008 from AE004092
SPy_0010  PSE-Membrane        tmhmm(1)                  SPy_0010 from AE004092
SPy_0012  PSE-Cellwall        hmm(GW2|GW3|GW1);signalp  SPy_0012 from AE004092
SPy_0016  MEMBRANE(non-PSE)   tmhmm(12)                 SPy_0016 from AE004092
SPy_0019  SECRETED            signalp                   SPy_0019 from AE004092
</code></pre><p>There are 4 columns to the result: 
</p>
<ol>
 <li>
     The sequence identifier (SeqID) is the first word in the id line of the FASTA file
 </li>

 <li>
     The cell-localization category:<ul>
 <li>
     PSE: means Potentially Surface Exposed, this can arise from<ul>
 <li>
     transmembrane helical protein with long extracellular loops
 </li>

 <li>
     lipoprotein attached to the membrane
 </li>

 <li>
     protein with motif that associates with the glycan cell-wall
 </li>
</ul>

 </li>

 <li>
     SECRETED: contains a secretion signal, but not the features above
 </li>

 <li>
     CYTOPLASM: contains none of the above features
 </li>
</ul>

 </li>

 <li>
     Summary of the results from the external bioinformatic programs. In this case, we have hits from <code>tmhmm</code>, <code>hmm</code>, and <code>signalp</code>
 </li>

 <li>
     The description of the protein taken from the FASTA file.
 </li>
</ol>


        </div>
      </div>

      <hr>

      <div class="footer">
        <p>&copy; 2012, Andrew Perry and Bosco Ho</p>
      </div>

    </div> <!-- /container -->

    <!-- Le javascript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="js/jquery.js"></script>
    <script src="js/bootstrap-transition.js"></script>
    <script src="js/bootstrap-alert.js"></script>
    <script src="js/bootstrap-modal.js"></script>
    <script src="js/bootstrap-dropdown.js"></script>
    <script src="js/bootstrap-scrollspy.js"></script>
    <script src="js/bootstrap-tab.js"></script>
    <script src="js/bootstrap-tooltip.js"></script>
    <script src="js/bootstrap-popover.js"></script>
    <script src="js/bootstrap-button.js"></script>
    <script src="js/bootstrap-collapse.js"></script>
    <script src="js/bootstrap-carousel.js"></script>
    <script src="js/bootstrap-typeahead.js"></script>

  </body>
</html>
